acme/src/authz.rs

```diff
@@ -93,39 +93,38 @@ impl AcmeAuthz {
     pub fn verify(&self) -> RustyAcmeResult<()> {
         let [challenge] = &self.challenges;
 
-        if let (AcmeIdentifier::WireappUser(_), AcmeChallengeType::WireDpop01)
-        | (AcmeIdentifier::WireappDevice(_), AcmeChallengeType::WireOidc01) = (&self.identifier, challenge.typ)
-        {
-            return Err(AcmeAuthzError::InvalidChallengeType)?;
-        };
-
-        let now = time::OffsetDateTime::now_utc().unix_timestamp();
-
-        let is_expired = self
-            .expires
-            .map(time::OffsetDateTime::unix_timestamp)
-            .map(|expires| expires < now)
-            .unwrap_or_default();
-        if is_expired {
-            return Err(AcmeAuthzError::Expired)?;
-        }
-
-        // RFC 8555 security considerations
-        let [ref challenge] = self.challenges;
+        if matches!(
+            (&self.identifier, challenge.typ),
+            (AcmeIdentifier::WireappDevice(_), AcmeChallengeType::WireDpop01)
+                | (AcmeIdentifier::WireappUser(_), AcmeChallengeType::WireOidc01)
+        ) {
+            let now = time::OffsetDateTime::now_utc().unix_timestamp();
+
+            let is_expired = self
+                .expires
+                .map(time::OffsetDateTime::unix_timestamp)
+                .map(|expires| expires < now)
+                .unwrap_or_default();
```

Comment:
```suggestion
                .is_some_and(|expires| expires < now);
```

Reply:
this is minor... feel free to just ignore it if you want

Reply:
Actually, I quite like your `is_some_and` suggestion, reads much more nicely. Thanks. :+1: